script.go

```diff
@@ -848,14 +851,16 @@ func (p *Pipe) Tee(writers ...io.Writer) *Pipe {
 	return p.WithReader(io.TeeReader(p.Reader, teeWriter))
 }
 
-// Wait reads the pipe to completion and discards the result. This is mostly
-// useful for waiting until concurrent filters have completed (see
-// [Pipe.Filter]).
-func (p *Pipe) Wait() {
+// Wait reads the pipe to completion and returns any error present on
+// the pipe, or nil otherwise. This is mostly useful for waiting until
+// concurrent filters have completed (see [Pipe.Filter]).
+func (p *Pipe) Wait() error {
 	_, err := io.Copy(io.Discard, p)
 	if err != nil {
 		p.SetError(err)
 	}
+
+	return p.Error()
```

Comment:
Is there a difference with doing this?


```suggestion
 		p.SetError(err)
 		return err
 	}

	return nil
```

It would be more readable

Reply:
i think currently there would be no difference but its nice to have a standard place for fetching the error in case we ever add anything to that method

Reply:
> Is there a difference with doing this?

But what if there's _already_ some error on the _pipe_, but the `io.Copy` was successful? In that case we would return `nil`, losing the existing `p.Error()`.

Reply:
Oh good catch